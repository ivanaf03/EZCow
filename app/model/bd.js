import * as SQLite from "expo-sqlite";

let db;

export const getDatabase = async () => {
  if (!db) {
    db = await SQLite.openDatabaseAsync("ezcow.db");
  }
  return db;
};

export const initDatabase = async () => {
  db = await getDatabase();

  await db.execAsync(`
        PRAGMA journal_mode = WAL;

        CREATE TABLE IF NOT EXISTS User (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            email TEXT NOT NULL UNIQUE,
            password TEXT DEFAULT NULL,
            google_id TEXT DEFAULT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            CHECK (password IS NOT NULL OR google_id IS NOT NULL)
        );

        CREATE TABLE IF NOT EXISTS Cow (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            code TEXT NOT NULL UNIQUE,
            name TEXT NOT NULL,
            entryDate DATE NOT NULL,
            exitDate DATE DEFAULT NULL,
            gender TEXT NOT NULL,
            breed TEXT NOT NULL,
            phase TEXT NOT NULL,
            user_fk INTEGER NOT NULL,
            mother_fk INTEGER DEFAULT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id),
            FOREIGN KEY(mother_fk) REFERENCES Cow(id)
        );

        CREATE TABLE IF NOT EXISTS Field (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            user_fk INTEGER NOT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id)
        );

        CREATE TABLE IF NOT EXISTS BreedingEvent (
            if INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            description TEXT,
            date DATE NOT NULL,
            cow_fk INTEGER NOT NULL,
            FOREIGN KEY(cow_fk) REFERENCES Cow(id)
        );
    `);
};

export const deleteDatabase = async () => {
  await db.closeAsync();
  await SQLite.deleteDatabaseAsync("ezcow.db");
};

export const closeDatabase = async () => {
  await db.closeAsync();
  db = null;
};

export const syncDatabase = async () => {
  await db.execAsync(`
        PRAGMA wal_checkpoint(FULL);
    `);
};
