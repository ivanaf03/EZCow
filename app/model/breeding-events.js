import { getDatabase } from "./bd";

export const insertBreedingEvent = async (
  cow_fk,
  description,
  date,
  user_fk
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO BreedingEvent (cow_fk, description, date, user_fk) VALUES (?, ?, ?, ?)`,
      [cow_fk, description, date, user_fk]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getBreedingEventsByDayAnUserId = async (day, userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM BreedingEvent JOIN Cow ON BreedingEvent.cow_fk = Cow.id WHERE date = ? AND user_fk = ?`,
      [userId, day]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};