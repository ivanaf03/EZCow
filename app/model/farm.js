import { getDatabase } from "./bd";

export const insertFarmUbication = async (userId, latitude, longitude) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `UPDATE User SET latitude = ?, longitude = ? WHERE id = ?`,
      [latitude, longitude, userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFarmUbicationByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT latitude, longitude FROM User WHERE id = ?`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
