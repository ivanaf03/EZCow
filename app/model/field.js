import { getDatabase } from './bd';

export const insertField = async (name, latitude, longitude, user_fk) => {

    db = await getDatabase();

    try {
        await db.runAsync(
            `INSERT INTO Field (name, latitude, longitude, user_fk) VALUES (?, ?, ?, ?)`,
            [name, latitude, longitude, user_fk]
        );
    } catch (error) {
        console.log(error);
    }
};

export const getAllFieldsByUserId = async (userId) => {
    db = await getDatabase();

    try {
        const res = await db.getAllAsync(
            `SELECT * FROM Field WHERE user_fk = ?`,
            [userId]
        );

        return res;
    } catch (error) {
        console.log(error);
    }
};