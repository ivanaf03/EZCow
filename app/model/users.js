import CryptoJS from "crypto-js";
import { getDatabase } from "./bd";

const hashPassword = (password) => {
  return CryptoJS.SHA256(password).toString();
};

export const insertUser = async (name, email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `INSERT INTO User (name, email, password) VALUES (?, ?, ?)`,
      [name, email, hashedPassword]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertUserGoogle = async (name, email, google_id) => {
  const db = await getDatabase();
  try {
    const existingUser = await db.getFirstAsync(
      `SELECT * FROM User WHERE email = ?`,
      [email]
    );
    if (existingUser) {
      return existingUser;
    }
    const res = await db.runAsync(
      `INSERT INTO User (name, email, google_id) VALUES (?, ?, ?)`,
      [name, email, google_id]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getUserByEmail = async (email) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(`SELECT * FROM User WHERE email = ?`, [
      email,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const changePassword = async (email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `UPDATE User SET password = ? WHERE email = ?`,
      [hashedPassword, email]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
