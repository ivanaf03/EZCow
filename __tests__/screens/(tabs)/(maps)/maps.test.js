import React from "react";

import { render, waitFor, fireEvent } from "@testing-library/react-native";
import { router } from "expo-router";

import Maps from "../../../../app/(tabs)/(maps)/maps";
import { getFarmUbicationByUserId, insertFarmUbication } from "../../../../model/farm";

jest.mock("../../../../store/user-provider", () => ({
    useUser: jest.fn(() => ({
        user: {id: "1", name: "TestUser", email: "test@test.com"},
    })),
}));

jest.mock("react-native-maps", () => {
    const React = require("react");
    const { View } = require("react-native");

    return {
        __esModule: true,
        default: (props) => React.createElement(View, props, props.children),
        Marker: (props) => React.createElement(View, props, props.children),
    };
});

jest.mock("../../../../model/farm", () => ({
    getFarmUbicationByUserId: jest.fn(), 
    insertFarmUbication: jest.fn(),
}));

jest.mock("expo-router", () => ({
    router: {
        replace: jest.fn(),
    },
}));

describe("Maps", () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("should render correctly", async () => {
        getFarmUbicationByUserId.mockResolvedValue({latitude: 0, longitude: 0});
        const tree = render(<Maps />);
        await waitFor(() => expect(tree).toMatchSnapshot());
    });

    it("should save correct location", async () => {
        getFarmUbicationByUserId.mockResolvedValue({latitude: 0, longitude: 0});
        const tree = render(<Maps />);
        await waitFor(() => expect(tree).toMatchSnapshot());
        const saveLocationButton = tree.getByText("Guardar ubicaciÃ³n");
        fireEvent.press(saveLocationButton);
        await waitFor(() => expect(insertFarmUbication).toHaveBeenCalledWith("1", 0, 0));
    });

    it("should navigate to fields", async () => {
        getFarmUbicationByUserId.mockResolvedValue({latitude: 0, longitude: 0});
        const tree = render(<Maps />);
        await waitFor(() => expect(tree).toMatchSnapshot());
        const fieldsButton = tree.getByTestId("fields-button");
        fireEvent.press(fieldsButton);
        await waitFor(() => expect(router.replace).toHaveBeenCalledWith("fields"));
    });
});