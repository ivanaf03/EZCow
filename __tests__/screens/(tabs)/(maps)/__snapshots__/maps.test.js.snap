// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps should navigate to fields 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 8,
        },
        {
          "paddingTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#ffffff",
          "borderLeftColor": "#ffffff",
          "borderRightColor": "#ffffff",
          "borderTopColor": "#ffffff",
        },
      ]
    }
  >
    Mapas
  </Text>
  <View
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#272727",
          },
          {
            "borderRightWidth": 8,
          },
          {
            "borderLeftWidth": 8,
          },
          {
            "borderBottomColor": "#2254ba",
            "borderLeftColor": "#2254ba",
            "borderRightColor": "#2254ba",
            "borderTopColor": "#2254ba",
          },
          {
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "marginBottom": 8,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
        ]
      }
      testID="fields-button"
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Nunito-Medium",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ]
        }
      >
        Ver mis fincas
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        {
          "backgroundColor": "#272727",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#6bc0eb",
          },
          {
            "fontSize": 20,
            "lineHeight": 28,
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
    >
      ◆ Selecciona la ubicación de tu explotación
    </Text>
    <Text
      style={
        [
          {
            "marginTop": 8,
          },
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "fontFamily": "Nunito-Italic",
          },
        ]
      }
    >
      Aquí puedes guardar la ubicación de tu explotación en el mapa. Se usará para centrar el mapa de las fincas de tu explotación.
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "backgroundColor": "#272727",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#6bc0eb",
          "borderLeftColor": "#6bc0eb",
          "borderRightColor": "#6bc0eb",
          "borderTopColor": "#6bc0eb",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "marginTop": 16,
        },
      ]
    }
  >
    <View
      onPress={[Function]}
      region={
        {
          "latitude": 0,
          "latitudeDelta": 0.01,
          "longitude": 0,
          "longitudeDelta": 0.01,
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        coordinate={
          {
            "latitude": 0,
            "longitude": 0,
          }
        }
        title="Tu explotación"
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        {
          "justifyContent": "space-around",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Latitud: 
      0.000000
    </Text>
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Longitud: 
      0.000000
    </Text>
  </View>
  <View
    style={
      [
        {
          "marginLeft": 64,
          "marginRight": 64,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#272727",
          },
          {
            "borderRightWidth": 8,
          },
          {
            "borderLeftWidth": 8,
          },
          {
            "borderBottomColor": "#2254ba",
            "borderLeftColor": "#2254ba",
            "borderRightColor": "#2254ba",
            "borderTopColor": "#2254ba",
          },
          {
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "marginBottom": 8,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Nunito-Medium",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ]
        }
      >
        Guardar ubicación
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Maps should render correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 8,
        },
        {
          "paddingTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#ffffff",
          "borderLeftColor": "#ffffff",
          "borderRightColor": "#ffffff",
          "borderTopColor": "#ffffff",
        },
      ]
    }
  >
    Mapas
  </Text>
  <View
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#272727",
          },
          {
            "borderRightWidth": 8,
          },
          {
            "borderLeftWidth": 8,
          },
          {
            "borderBottomColor": "#2254ba",
            "borderLeftColor": "#2254ba",
            "borderRightColor": "#2254ba",
            "borderTopColor": "#2254ba",
          },
          {
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "marginBottom": 8,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
        ]
      }
      testID="fields-button"
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Nunito-Medium",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ]
        }
      >
        Ver mis fincas
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        {
          "backgroundColor": "#272727",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#6bc0eb",
          },
          {
            "fontSize": 20,
            "lineHeight": 28,
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
    >
      ◆ Selecciona la ubicación de tu explotación
    </Text>
    <Text
      style={
        [
          {
            "marginTop": 8,
          },
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "fontFamily": "Nunito-Italic",
          },
        ]
      }
    >
      Aquí puedes guardar la ubicación de tu explotación en el mapa. Se usará para centrar el mapa de las fincas de tu explotación.
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "backgroundColor": "#272727",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#6bc0eb",
          "borderLeftColor": "#6bc0eb",
          "borderRightColor": "#6bc0eb",
          "borderTopColor": "#6bc0eb",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "marginTop": 16,
        },
      ]
    }
  >
    <View
      onPress={[Function]}
      region={
        {
          "latitude": 0,
          "latitudeDelta": 0.01,
          "longitude": 0,
          "longitudeDelta": 0.01,
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        coordinate={
          {
            "latitude": 0,
            "longitude": 0,
          }
        }
        title="Tu explotación"
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        {
          "justifyContent": "space-around",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Latitud: 
      0.000000
    </Text>
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Longitud: 
      0.000000
    </Text>
  </View>
  <View
    style={
      [
        {
          "marginLeft": 64,
          "marginRight": 64,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#272727",
          },
          {
            "borderRightWidth": 8,
          },
          {
            "borderLeftWidth": 8,
          },
          {
            "borderBottomColor": "#2254ba",
            "borderLeftColor": "#2254ba",
            "borderRightColor": "#2254ba",
            "borderTopColor": "#2254ba",
          },
          {
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "marginBottom": 8,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Nunito-Medium",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ]
        }
      >
        Guardar ubicación
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Maps should save correct location 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "marginTop": 8,
        },
        {
          "paddingTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 36,
          "lineHeight": 40,
        },
        {
          "fontFamily": "Nunito-Bold",
        },
        {
          "borderBottomWidth": 2,
        },
        {
          "borderBottomColor": "#ffffff",
          "borderLeftColor": "#ffffff",
          "borderRightColor": "#ffffff",
          "borderTopColor": "#ffffff",
        },
      ]
    }
  >
    Mapas
  </Text>
  <View
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#272727",
          },
          {
            "borderRightWidth": 8,
          },
          {
            "borderLeftWidth": 8,
          },
          {
            "borderBottomColor": "#2254ba",
            "borderLeftColor": "#2254ba",
            "borderRightColor": "#2254ba",
            "borderTopColor": "#2254ba",
          },
          {
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "marginBottom": 8,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
        ]
      }
      testID="fields-button"
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Nunito-Medium",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ]
        }
      >
        Ver mis fincas
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "marginTop": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        {
          "backgroundColor": "#272727",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#6bc0eb",
          },
          {
            "fontSize": 20,
            "lineHeight": 28,
          },
          {
            "fontFamily": "Nunito-Medium",
          },
        ]
      }
    >
      ◆ Selecciona la ubicación de tu explotación
    </Text>
    <Text
      style={
        [
          {
            "marginTop": 8,
          },
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "fontFamily": "Nunito-Italic",
          },
        ]
      }
    >
      Aquí puedes guardar la ubicación de tu explotación en el mapa. Se usará para centrar el mapa de las fincas de tu explotación.
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "backgroundColor": "#272727",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#6bc0eb",
          "borderLeftColor": "#6bc0eb",
          "borderRightColor": "#6bc0eb",
          "borderTopColor": "#6bc0eb",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "marginTop": 16,
        },
      ]
    }
  >
    <View
      onPress={[Function]}
      region={
        {
          "latitude": 0,
          "latitudeDelta": 0.01,
          "longitude": 0,
          "longitudeDelta": 0.01,
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <View
        coordinate={
          {
            "latitude": 0,
            "longitude": 0,
          }
        }
        title="Tu explotación"
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        {
          "justifyContent": "space-around",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Latitud: 
      0.000000
    </Text>
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Longitud: 
      0.000000
    </Text>
  </View>
  <View
    style={
      [
        {
          "marginLeft": 64,
          "marginRight": 64,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "backgroundColor": "#272727",
          },
          {
            "borderRightWidth": 8,
          },
          {
            "borderLeftWidth": 8,
          },
          {
            "borderBottomColor": "#2254ba",
            "borderLeftColor": "#2254ba",
            "borderRightColor": "#2254ba",
            "borderTopColor": "#2254ba",
          },
          {
            "borderBottomLeftRadius": 24,
            "borderBottomRightRadius": 24,
            "borderTopLeftRadius": 24,
            "borderTopRightRadius": 24,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "marginBottom": 8,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
            },
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "fontFamily": "Nunito-Medium",
            },
            {
              "marginLeft": 8,
              "marginRight": 8,
            },
          ]
        }
      >
        Guardar ubicación
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
